src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/configs/CacheActionConfigTest.java

```diff
@@ -105,4 +105,48 @@ public void testInvalidCacheType() throws Exception {
     CacheActionConfig cacheActionConfig = new CacheActionConfig(conf);
     cacheActionConfig.getThresholdConfig(ResourceEnum.SEARCH_THREADPOOL).upperBound();
   }
+
+  @Test
+  public void testGetStepSize() throws Exception {
+    String configStr =
+        "{"
+          + "\"action-config-settings\": { "
+              + "\"cache-settings\": { "
+                  + "\"fielddata\": { "
+                      + "\"upper-bound\": 0.8, "
+                      + "\"lower-bound\": 0.2 "
+                  + "}, "
+                  + "\"shard-request\": { "
+                      + "\"upper-bound\": 0.08, "
+                      + "\"lower-bound\": 0.01 "
+                  + "} "
+              + "} "
+          + "} "
+      + "}";
+    RcaConf conf = new RcaConf();
+    conf.readConfigFromString(configStr);
+    CacheActionConfig cacheActionConfig = new CacheActionConfig(conf);
+    assertEquals(0.03, cacheActionConfig.getStepSize(ResourceEnum.FIELD_DATA_CACHE), 0.0001);
+    assertEquals(0.0035, cacheActionConfig.getStepSize(ResourceEnum.SHARD_REQUEST_CACHE), 0.0001);
+    configStr =
+        "{"
+          + "\"action-config-settings\": { "
+              + "\"total-step-count\": 5,"
```

Comment:
As per your comment 
```step count is indeed part of the cacheSettingConfig. But here we need to pass the parent Map<> object as a parameter in the constructor of the Config class```
In that case this step count might need to be inside the cache-settings block right?